docs/examples/test_examples.py

```diff
@@ -0,0 +1,268 @@
+""" Tests for the files in nameko/docs/examples
+"""
+import os
+
+import eventlet
+from mock import call, patch
+import pytest
+import requests
+
+from nameko.containers import ServiceContainer
+from nameko.standalone.events import event_dispatcher
+from nameko.standalone.rpc import ServiceRpcProxy
+from nameko.testing.services import entrypoint_waiter
+from nameko.testing.websocket import make_virtual_socket
+
+
+@pytest.yield_fixture
+def container_factory(rabbit_config):
+
+    all_containers = []
+
+    def make_container(service_cls, config, worker_ctx_cls=None):
+        container = ServiceContainer(service_cls, config, worker_ctx_cls)
+        all_containers.append(container)
+        return container
+
+    yield make_container
+
+    for c in all_containers:
+        try:
+            c.stop()
+        except:
+            pass
+
+
+@pytest.yield_fixture()
+def websocket():
+    active_sockets = []
+
+    def socket_creator():
+        ws_app, wait_for_sock = make_virtual_socket(
+            '127.0.0.1', 8000)
```

Comment:
in the other web tests we jump through hoops to make sure we pick a random (free) port